feat: add DesignTokens type for managing app Design Tokens (#4411)
microsoft/fast: packages/web-components/fast-foundation/src/design-token/README.md

**This guide describes APIs and systems that are alpha and subject to change. Please keep that in mind before using these tools. If you have feedback, please don't hesitate to file an issue!**

# Design Tokens
The FAST Design Token implementation is designed to provide first-class support for Design Tokens and make setting, getting, and using Design Tokens simple.

## What is a Design Token
A Design Token is a semantic, named variable used to describe a Design System. They often describe design concepts like typography, color, sizes, UI spacing, etc. FAST encourages checking out [https://github.com/design-tokens/community-group#design-tokens](the Design Tokens Community Group) for more information on Design Tokens themselves.

## Using Design Tokens in FAST
This usage guide walks through creating and using a Design Token *accent-color*, which is represented in code as a hexadecimal color string.

### 1. Create a Token
The first step to using a token is to create it:

```ts
import { DesignToken } from "@microsoft/fast-foundation";

export const accentColor = DesignToken.create<string>("accent-color");
```

> The type assertion informs why types the token can be set to (and what type will be retrieved), and the name parameter will serve as the CSS Custom Property name (more on that later).

Reviewer: ```suggestion
The type assertion informs what types the token can be set to (and what type will be retrieved), and the name parameter will serve as the CSS Custom Property name (more on that later).
```

### 2. Setting the Design Token
A `DesignToken` *value* is set for a `FASTElement` node. This allows tokens to be set to different values for distinct DOM trees:

```ts
const ancestor = document.querySelector("my-element") as FASTElement & HTMLElement;
const descendent = ancestor.querySelector("my-element")

accentColor.setValueFor(ancestor, "#DA1A5F");

> descendent.setValueFor(descendent, "#ADA1F5");

Reviewer: ```suggestion
accentColor.setValueFor(descendent, "#ADA1F5");
```

 ```

### 3. Getting the Design Token value
Once the value is set for a node, the value is available to use for that node or any descendent node. The value returned will be the value set for the nearest ancestor (or the element itself).

```ts
accentColor.getValueFor(ancestor); // "#DA1A5F"
accentColor.getValueFor(descendent); // "#ADA1F5"
```

### 4. Deleting Design Token values
Values can be deleted for a node. Doing so causes retrieval of the nearest ancestors value instead:

```ts
accentColor.deleteValueFor(descendent);
accentColor.getValueFor(descendent); // "#DA1A5F"
```

### 5. Emit a token to a CSS Custom Property

> A Design Token can be made available in CSS through CSS custom properties. The custom property value be set to the tokens value for the supplied target element.

Reviewer: ```suggestion
A Design Token can be made available in CSS through CSS custom properties. The custom property value will be set to the token's value for the supplied target element.
```
